MnemonicSwift/Mnemonic.swift

```diff
@@ -93,51 +134,94 @@ public enum Mnemonic {
+            guard SecRandomCopyBytes(kSecRandomDefault, count, &bytes) == errSecSuccess else {
+                throw MnemonicError.entropyCreationFailed
+            }
 
-            return mnemonicString(from: bytes.hexString, language: language)
+            return try mnemonicString(from: bytes.hexString, language: language)
     }
 
-    /// Validate that the given string is a valid mnemonic.
-    public static func validate(mnemonic: String) -> Bool {
-        let normalizedMnemonic = mnemonic.lowercased().trimmingCharacters(in: .whitespacesAndNewlines)
+    /// Validate that the given string is a valid mnemonic phrase according to BIP-39
+    /// - Parameters:
+    ///  - mnemonic: a mnemonic phrase string
+    /// - Throws:
+    ///  - `MnemonicError.wrongWordCount` if the word count is invalid
+    ///  - `MnemonicError.invalidWord(word: word)` this phase as a word that's not represented in this library's vocabulary for the detected language.
+    ///  - `MnemonicError.unsupportedLanguage` if the given phrase language isn't supported or couldn't be infered
+    ///  - `throw MnemonicError.checksumError` if the given phrase has an invalid checksum
+    public static func validate(mnemonic: String) throws {
+        let normalizedMnemonic = mnemonic.trimmingCharacters(in: .whitespacesAndNewlines)
         let mnemonicComponents = normalizedMnemonic.components(separatedBy: " ")
         guard !mnemonicComponents.isEmpty else {
-            return false
+            throw MnemonicError.wrongWordCount
         }
 
-        // Use the first component of the mnemonic to determine the language, then make sure all
-        // subsequent components are in that language.
-        if String.englishMnemonics.contains(mnemonicComponents[0]) {
-            for mnemonicComponent in mnemonicComponents {
-                guard String.englishMnemonics.contains(mnemonicComponent) else {
-                    return false
-                }
-            }
-            return true
-        } else if String.chineseMnemonics.contains(mnemonicComponents[0]) {
-            for mnemonicComponent in mnemonicComponents {
-                guard String.chineseMnemonics.contains(mnemonicComponent) else {
-                    return false
-                }
+        // determine the language of the seed or fail
+        let language = try determineLanguage(from: mnemonicComponents)
+        let vocabulary = language.words()
+
+        // generate indices array
+        var seedBits = ""
+        for word in mnemonicComponents {
+            guard let indexInVocabulary = vocabulary.firstIndex(of: word) else {
+                throw MnemonicError.invalidWord(word: word)
             }
-            return true
+
+            let binaryString = String(indexInVocabulary, radix: 2).pad(toSize: 11)
+
+            seedBits.append(contentsOf: binaryString)
+        }
+
+        let checksumLength = mnemonicComponents.count / 3
```

Comment:
We should check that `mnemonicComponents.count` is valid according to the enum options in `WordCount`. I think this will accept any one-word or two-word "mnemonmic" as valid, because `2 / 3 = 0` and so it will try to check a 0-length checksum which will always pass.

Add a test for this, e.g. "flash tobacco". Currently, I think that will pass validation when it shouldn't.